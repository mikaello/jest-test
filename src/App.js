import React, { Component } from 'react'
import { connect } from 'react-redux'

import logo from './logo.svg'
import './App.css'

export const ListOfCrackers = ({ crackerList }) => (
  <div>
    <h4>Bøllene her på huset er:</h4>
    <ul>
      {crackerList.length > 0
        ? crackerList.map((cracker, index) => <li key={index}>{cracker}</li>)
        : 'Ingen bøller :-('}
    </ul>
  </div>
)

class App extends Component {
  addCracker() {
    return (
      <div>
        <h4>Legg til bølle:</h4>
        <input ref="newCracker" />
        <button
          type="submit"
          onClick={() => {
            this.props.blameBottleCracker(this.refs.newCracker.value)
            this.refs.newCracker.value = ''
          }}
        >
          Trykk for å legge til en BØLLE
        </button>
      </div>
    )
  }

  render() {
    return (
      <div className="App">
        <div className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h2>
            Velkommen til Jest workshop i React, sjefen er {this.props.theBoss}
          </h2>
        </div>

        <p className="App-intro">
          Please skaff en ny sjef:
        </p>
        <input
          onChange={boss => this.props.anotherBossPlease(boss.target.value)}
        />

        <ListOfCrackers crackerList={this.props.bottleCrackers} />

        {this.addCracker()}

      </div>
    )
  }
}

const ConnectedPage = connect(
  state => ({
    theBoss: state.sjefen,
    bottleCrackers: state.kolbeknusere,
  }),
  dispatch => ({
    anotherBossPlease: newBoss => dispatch({ type: 'KICK_THE_BOSS', newBoss }),
    blameBottleCracker: bottleCracker =>
      dispatch({ type: 'BLAME_CRACKER', bottleCracker }),
  })
)(App)

export default ConnectedPage
