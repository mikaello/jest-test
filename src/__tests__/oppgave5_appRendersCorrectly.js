import React from 'react'
import renderer from 'react-test-renderer'
import { createStore } from 'redux'

import usitTasks from '../usitReducer'
import App from '../App'

test('app renders correctly', () => {
  const tree = renderer.create(<App store={createStore(usitTasks)} />)
  expect(tree).toMatchSnapshot()
})
