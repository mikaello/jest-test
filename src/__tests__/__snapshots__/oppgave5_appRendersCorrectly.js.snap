// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h2>
      Velkommen til Jest workshop i React, sjefen er 
      OFTIZ
    </h2>
  </div>
  <p
    className="App-intro"
  >
    Please skaff en ny sjef:
  </p>
  <input
    onChange={[Function]}
  />
  <div>
    <h4>
      Legg til bølle:
    </h4>
    <input />
    <button
      onClick={[Function]}
      type="submit"
    >
      Trykk for å legge til en BØLLE
    </button>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h2>
      Velkommen til Jest workshop i React, sjefen er 
      OFTIZ
    </h2>
  </div>
  <p
    className="App-intro"
  >
    Please skaff en ny sjef:
  </p>
  <input
    onChange={[Function]}
  />
  <div>
    <h4>
      Legg til bølle:
    </h4>
    <input />
    <button
      onClick={[Function]}
      type="submit"
    >
      Trykk for å legge til en BØLLE
    </button>
  </div>
</div>
`;
